internal/storage/rocksdb/store.go

```diff
@@ -515,7 +592,17 @@ func (rdb *rocksDB) getSingleKey(ro *gorocksdb.ReadOptions, key []byte) ([]*serv
 	defer value.Free()
 	val := toByteArray(value)
 	if val != nil && len(val) > 0 {
-		return []*serverpb.KVPair{&serverpb.KVPair{Key: key, Value: val}}, nil
+		//check ttl.
+		expireTs, _ := getExpireTs(val)
+		if expireTs > 0 && expireTs < time.Now().Unix() {
+			//trigger delete.
+			rdb.Delete(key)
+			return nil, nil
+		} else if expireTs > 0 {
+			val = val[0 : len(val)-tsLength]
```

Comment:
What if the value is a genuine timestamp ?

Reply:
Byte marker introduced which allows a int64 max of `72057594037927935` (7 bit data) and reserves (1 bit for marker)

Reply:
Split into separate CF for values with TS.